src/integration-tests/src/bigquery.rs

```diff
@@ -0,0 +1,145 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Result;
+use gax::error::Error;
+use rand::{Rng, distr::Alphanumeric};
+
+pub async fn dataset_admin(
+    builder: bigquery::builder::dataset_service::ClientBuilder,
+) -> Result<()> {
+    // Enable a basic subscriber. Useful to troubleshoot problems and visually
+    // verify tracing is doing something.
+    #[cfg(feature = "log-integration-tests")]
+    let _guard = {
+        use tracing_subscriber::fmt::format::FmtSpan;
+        let subscriber = tracing_subscriber::fmt()
+            .with_level(true)
+            .with_thread_ids(true)
+            .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
+            .finish();
+
+        tracing::subscriber::set_default(subscriber)
+    };
+
+    let project_id = crate::project_id()?;
+    let client = builder.build().await?;
+    cleanup_stale_datasets(&client, &project_id).await?;
+
+    let rand_suffix: String = rand::rng()
+        .sample_iter(&Alphanumeric)
+        .take(8)
+        .map(char::from)
+        .collect();
+
+    let ds_name = format!("rust_bq_test_dataset_{rand_suffix}");
+
+    println!("CREATING DATASET WITH ID: {ds_name}");
+
+    let create = client
+        .insert_dataset(&project_id)
+        .set_dataset(bigquery::model::Dataset::new().set_dataset_reference(
+            bigquery::model::DatasetReference::new().set_dataset_id(&ds_name),
+        ))
+        .send()
+        .await?;
+    println!("CREATE DATASET = {create:?}");
+
+    assert!(create.dataset_reference.is_some());
+
+    let list = client.list_datasets(&project_id).send().await?;
+    println!("LIST DATASET = {} entries", list.datasets.len());
+
+    assert!(!list.datasets.is_empty());
+    assert!(list.datasets.len() > 1);
+    assert!(
+        list.datasets
+            .iter()
+            .find(|v| v.id.contains(&ds_name))
+            .is_some()
+    );
+
+    let delete = client
+        .delete_dataset(&project_id, &ds_name)
+        .set_delete_contents(true)
+        .send()
+        .await?;
+    println!("DELETE DATASET = {delete:?}");
+
+    //assert!(delete().is_ok());
+
+    Ok(())
+}
+
+async fn cleanup_stale_datasets(
+    client: &bigquery::client::DatasetService,
+    project_id: &str,
+) -> Result<()> {
+    use std::time::{Duration, SystemTime, UNIX_EPOCH};
+    let stale_deadline = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .map_err(Error::other)?;
+    let stale_deadline = stale_deadline - Duration::from_secs(48 * 60 * 60);
+    let stale_deadline = stale_deadline.as_millis() as i64;
+
+    let list = client.list_datasets(project_id).send().await?;
+    let pending_all_datasets = list
+        .datasets
+        .iter()
+        .map(|v| {
+            return client
+                .get_dataset(
+                    project_id,
+                    v.dataset_reference.as_ref().map_or("", |v| &v.dataset_id),
```

Comment:
This deserves some comment explaining what is going on.  If the `dataset_reference` is not set it seems we cannot make a successful call to `.get_dataset()`?  Maybe returning `None` is more appropriate? or maybe using `v.id` and stripping the project_id?

Maybe we should have something like:

```rust
fn extract_dataset_id(project_id: &str, v: bigquery::model::ListFormatDataset) -> Option<String> {
    match v.dataset_reference {
        Some(r) => r.dataset_id.clone,
        None => v.id.strip_prefix("projects/").strip_prefix(project_id).owned(),
    }
}
```

and then we issue the request only if we have `Some()`...